src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerTests.cs

```diff
@@ -29,14 +36,57 @@ public override int GetStakeMinConfirmations(int height, Network network)
             }
         }
 
+        public class StratisOverrideRegTest : StratisRegTest
+        {
+            public StratisOverrideRegTest() : base()
+            {
+                this.Name = Guid.NewGuid().ToString();
+            }
+        }
+
+        public class BitcoinOverrideRegTest : BitcoinRegTest
+        {
+            public BitcoinOverrideRegTest() : base()
+            {
+                this.Name = Guid.NewGuid().ToString();
+            }
+        }
```

Comment:
Rather then create override type classes, can't the derived classes include an extra ctor that passes in/sets Name ?

Otherwise any future tests would need to follow this code pattern just to set the Name.

Reply:
we want to keep Network as immutable as possible ;)